plugins/org/src/components/MyGroups/MyGroups.tsx

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import {
+  SidebarItem,
+  SidebarSubmenu,
+  SidebarSubmenuItem,
+} from '@backstage/core-components';
+import {
+  IconComponent,
+  identityApiRef,
+  useApi,
+} from '@backstage/core-plugin-api';
+import useAsync from 'react-use/lib/useAsync';
+import { catalogApiRef, CatalogApi } from '@backstage/plugin-catalog-react';
+
+export const MyGroups = ({
+  singularTitle,
+  pluralTitle,
+  icon,
+}: {
```

Comment:
If you make this just `(props: {...`, the exported api report will be much neater. Then you can do the unpacking of these inside the component body instead. I know the rest of the plugin isn't pretty in terms of the api report, but let's start doing good!

While you're at it, add a doc comment ```/**``` including a `@public` so it's clear that it's for public use. That description is also a good place to mention that this is for sidebar use - in a couple months we're likely to have forgotten what it's for. Should probably even name it `MyGroupsSidebarItem`.

Reply:
I've made these changes but I can't seem to figure out how to get the description to show up in the API Report. Any ideas?

Reply:
That would be three props and that's what we want!

So look at these two functions

```ts
function a1(props: T) {
  const { foo, bar } = props;
}
function a2({ foo, bar }: T) {
}
```

I removed the curlies on the right side of the colon because they don't matter to this discussion; I just named that bit T for now.

`a1` and `a2` both have ONE argument. It's just that in a1, the function body has access to a variable `props` (which we unpack to grab its fields foo and bar inside); while in a2, we use the shorthand to unpack it directly (the name `props` never enters the picture in the first place).

But they are completely functionally identical.

We do NOT like the a2 form where the argument is unpacked directly. It's confusing and the api report gets really hard to read. We prefer that the argument just gets a human readable name - `props` which everybody knows what it means and clearly conveys that it's a component - and then do the unpacking inside the function body instead. As a1 does.

The a2 early unpacking is for sure a point of confusion for newcomers too. As you've felt! It's a pretty recent modern feature of js and it trips people up. We've been doing things that way a lot too but are actively trying to get away from it and change to the other form as we go along.

Reply:
And to be clear, in both of those cases you'd render them as `<a* foo={} bar={} />`. Two props, but always ONE argument to the function (which is an object with two fields), also named "props".

Sorry bout the lowercase component name, don't want to update the previous text now :)

Reply:
OK, I "think" I'm starting to understand this better. I'm getting caught up on the "T" part in the example. This is where I ended up:

```tsx
export type MyGroupsSidebarItemProps ={
  singularTitle: string,
  pluralTitle: string,
  icon: IconComponent
}

export const MyGroupsSidebarItem = (props: MyGroupsSidebarItemProps) => {
  const { singularTitle, pluralTitle, icon } = props;
```

Is this the right direction to go?